packages/jsii/lib/assembler.ts

```diff
@@ -1587,7 +1628,7 @@ export class Assembler implements Emitter {
       for (const member of declaringType.getProperties()) {
         if (
           !(declaringType.symbol.getDeclarations() ?? []).find(
-            (decl) => decl === member.valueDeclaration.parent,
+            (decl) => decl === member.valueDeclaration?.parent,
```

Comment:
The `?` here is due to `valueDeclaration` being nullable (despite the type declarations do not mention it).